### ISSUE TITLE
FrameworkSearchPath issue for Xcode Framework (ld: framework not found)

### ISSUE THREAD

**Issue by marcvanolmen** [CocoaPods/CocoaPods#4516]
Sorry haven't had the time to make a simple project to reproduce this issue:

But our iOS project has an iOS app target and an Xcode iOS Framework, we use Xcode to build and link this framework.

Both the app OurProdTV and the framework target have a list of cocoapods specified in the Podfile, one of them is our own private Pod (= TRAAC) that both have listed.

Cocoapod generates 2 Build Targets for each of these imports in the pods.projects

Pods-OurProd-TRAAC
Pods-OurProdCore-TRAAC

Traac itself has dependencies specified in its pod spec

  s.dependency 'Mixpanel',      '2.5.3'
  s.dependency 'Mantle',        '1.3.1'
  s.dependency 'Analytics/Mixpanel','1.7.9'

pod install decide to make now specific Mixpanel target for each of the our own Pods-targets.

But when we try to build we get Pods-OurProd-TRAAC or Pods-OurProdCore-TRAAC we get error:

ld: framework not found Mixpanel
clang: error: linker command failed with exit code 1 (use -v to see invocation)

If I go look a the build folder: 

Debug-iphonesimulator
     • Mixpanel.framework
     • Pods-OurProd
               • TRAAC.framework
     • Pods-OurProdCore
               • TRAAC.framework

When I look at the config file that cocoapods generated the the Path 'Debug-iphonesimulator' is actually not part of the search FRAMEWORK_SEARCH_PATHS.  just the path to the Pods-OurProd

```
FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_FRAMEWORK_BUILD_PATH"
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1 $(inherited) BITHOCKEY_VERSION="@\"3.7.1\"" BITHOCKEY_C_VERSION="\"3.7.1\"" BITHOCKEY_BUILD="@\"42\"" BITHOCKEY_C_BUILD="\"42\""
LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
OTHER_CFLAGS = $(inherited) -iquote "$CONFIGURATION_BUILD_DIR/Bolts.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/FBSDKCoreKit.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/FBSDKLoginKit.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/FXForms.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/HockeySDK_Source.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/Masonry.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/Mixpanel.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/PriorityPush.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/TTTAttributedLabel.framework/Headers" -iquote "$CONFIGURATION_BUILD_DIR/YLMoment.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/Aspects.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/FXKeychain.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/JLRoutes.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/MKTools.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/Mantle.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/Objection.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/RMStore.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/RNCryptor.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/RTVVideoEffects.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/SDWebImage.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/TRAAC.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/TRToolkit.framework/Headers" -iquote "$PODS_FRAMEWORK_BUILD_PATH/TRVideoKit.framework/Headers"
OTHER_LDFLAGS = $(inherited) -ObjC -all_load -framework "Aspects" -framework "Bolts" -framework "FBSDKCoreKit" -framework "FBSDKLoginKit" -framework "FXForms" -framework "FXKeychain" -framework "HockeySDK_Source" -framework "JLRoutes" -framework "MKTools" -framework "Mantle" -framework "Masonry" -framework "Mixpanel" -framework "Objection" -framework "PriorityPush" -framework "RMStore" -framework "RNCryptor" -framework "RTVVideoEffects" -framework "SDWebImage" -framework "TRAAC" -framework "TRToolkit" -framework "TRVideoKit" -framework "TTTAttributedLabel" -framework "YLMoment"
PODS_FRAMEWORK_BUILD_PATH = $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-OurProd
PODS_ROOT = ${SRCROOT}/Pods
``` 

here the detailed link error

```
Ld /Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Products/Debug-iphonesimulator/Pods-OurProd/TRAAC.framework/TRAAC normal x86_64
    cd /Users/marcvano/dev/OurProd/RTV_iOS_Dev/Pods
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.1.sdk -L/Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Products/Debug-iphonesimulator/Pods-OurProd -F/Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Products/Debug-iphonesimulator/Pods-OurProd -filelist /Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Intermediates/Pods.build/Debug-iphonesimulator/Pods-OurProd-TRAAC.build/Objects-normal/x86_64/TRAAC.LinkFileList -install_name @rpath/TRAAC.framework/TRAAC -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -mios-simulator-version-min=8.0 -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -framework Foundation -framework Mantle -framework Mixpanel -single_module -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Intermediates/Pods.build/Debug-iphonesimulator/Pods-OurProd-TRAAC.build/Objects-normal/x86_64/TRAAC_dependency_info.dat -o /Users/marcvano/Library/Developer/Xcode/DerivedData/OurProd-ajnyzuuihfuuvkfiukiualeqtvms/Build/Products/Debug-iphonesimulator/Pods-OurProd/TRAAC.framework/TRAAC
```

if I manually add the path, to the search path to FRAMEWORK_SEARCH_PATHS it works.

**Comment by segiddins** [comment]
Cocoapods doesn't yet support integrating with framework targets -- you'd have to make the framework itself a pod.

**Comment by orta** [comment]
Can we check the integration type and make it quit during `pod install` if they do that?

**Comment by segiddins** [comment]
@orta I'd rather we make it work, but either way a PR would be appreciated.

**Comment by neonichu** [comment]
I don't think we can really make it work, considering that we can't embed the dependencies into a framework.

**Comment by marcvanolmen** [comment]
We added more brainpower to this issue and we found some interesting things:

Show version:

The issue in our case gets fixed when we remove the 2 targets that are specific for tvOS platform from the pod file. I didn't mention this in my my initial explanation.

More Longer explanation:

When we do this:

```
Podfile:

target : Pods-OurProd do
    platform :ios, '8.0'
    pod 'TRAAC', '1.4'
end

# our framework
target : Pods-OurProdCore do
    platform :ios, '8.0'
    pod 'TRAAC', '1.4'
end

target : Pods-OurProd-TV do
    platform :tvos, '8.0'
    pod 'TRAAC', '1.4'
end

# our framework
target : Pods-OurProdCore-TV do
    platform :tvos, '8.0'
    pod 'TRAAC', '1.4'
end

```
When build the product result build folder looks like this:

- Mixpanel.framework
- Pods-OurProd
- - - - - -  TRAAC.framework
- Pods-OurProdCore
- - - - - -  TRAAC.framework

And gives a link error, because Traac can't find mixpanel.

In case we comment out the tvOS targets, the result folder looks like this:

- Mixpanel.framework
- TRAAC.framework

So TRAAC gets build only once, and more importantly places all the frameworks and apps at the same level. No more 2 subfolders.

If I look at the Targets now generated in the Prods.project:

- MixPanel
- Pods-OurProd
- Pods-OurProdCore
- Traac

Which is different then what I mentioned above. (Pods-OurProd-TRAAC, Pods-OurProdCore-TRAAC) for TRAAC.

So it seems that specifying an extra platform cause cocoapods to be confused.

**Comment by segiddins** [comment]
@marcvanolmen I think your issue will be fixed by https://github.com/CocoaPods/CocoaPods/pull/4146, could you check since you haven't shared a project with us? Thanks!

**Comment by segiddins** [comment]
Closing due to lack of response -- this should be fixed in 1.0.
